src/LanguageServer/Microsoft.CommonLanguageServerProtocol.Framework/RequestExecutionQueue.cs

```diff
@@ -290,24 +275,121 @@ private async Task ProcessQueueAsync()
     }
 
     /// <summary>
-    /// Allows XAML to inspect the request before its dispatched.
-    /// Should not generally be used, this will be replaced by the OOP XAML server.
+    /// Reflection invokes <see cref="ProcessQueueCoreAsync{TRequest, TResponse}(IQueueItem{TRequestContext}, IMethodHandler, RequestHandlerMetadata, ConcurrentDictionary{Task, CancellationTokenSource}, CancellationTokenSource?, CancellationToken)"/>
+    /// using the concrete types defined by the handler's metadata.
     /// </summary>
-    [Obsolete("Only for use by legacy XAML LSP")]
-    protected internal virtual void BeforeRequest<TRequest>(TRequest request)
+    private async Task InvokeProcessCoreAsync(
+        IQueueItem<TRequestContext> work,
+        RequestHandlerMetadata metadata,
+        IMethodHandler handler,
+        MethodInfo methodInfo,
+        ConcurrentDictionary<Task, CancellationTokenSource> concurrentlyExecutingTasks,
+        CancellationTokenSource? currentWorkCts,
+        CancellationToken cancellationToken)
     {
-        return;
+        var task = methodInfo.Invoke(this, [work, handler, metadata, concurrentlyExecutingTasks, currentWorkCts, cancellationToken]) as Task
+            ?? throw new InvalidOperationException($"ProcessQueueCoreAsync result task cannot be null");
```

Comment:
this exception is slightly wrong.  it will also fire if something returns a *non-task* result.  my pref is two throws.  The first if we get null back, the second if we can't convert to a task.

Reply:
note: if we're going to throw on `as` you shoudl just `cast` instead :)